syncer/checkpoint.go

```diff
@@ -375,8 +378,12 @@ func (cp *RemoteCheckPoint) DeleteSchemaPoint(tctx *tcontext.Context, sourceSche
 	return nil
 }
 
-// IsNewerTablePoint implements CheckPoint.IsNewerTablePoint
-func (cp *RemoteCheckPoint) IsNewerTablePoint(sourceSchema, sourceTable string, location binlog.Location) bool {
+// IsNewerTablePoint implements CheckPoint.IsNewerTablePoint.
+// gte means greater than or equal, gte should judge by EnableGTID and the event type
+// - when enable GTID and binlog is DML, different location may have same GTID, so when GTID is equal, can treat it as new table point
+// - when enable GTID and binlog is DDL, different DDL have different GTID, so if GTID is euqal, it is a old table point
+// - when not enable GTID, just compare the position, and only when grater than the old point is newer table point
```

Comment:
Because DML doesn't update table checkpoint?

Reply:
no, because go-mysql update GTID set of rows event in XID event after this rows event. I update the comment and add a example.